File: src/3396.minimum-number-of-operations-to-make-elements-in-array-distinct.3656/solution.ts
function minimumOperations (nums: number[]): number {
    const counts = new Array<number>(101).fill(0);
    let duplicate = 0;
    for (const n of nums) {
        counts[n]++;
        if (counts[n] === 2) {
            duplicate++;
        }
    }

    let result = 0;
    while (duplicate > 0) {
        for (let i = 3 * result; i < Math.min(nums.length, (result + 1) * 3); i++) {
            const n = nums[i];
            counts[n]--;
            if (counts[n] === 1) {
                duplicate--;
            }
        }

        result++;
    }

    return result;
}

File: src/3396.minimum-number-of-operations-to-make-elements-in-array-distinct.3656/solution.java
class Solution {
    public int minimumOperations(int[] nums) {
        int[] counts = new int[101];
        int duplicate = 0;
        for(int n:nums){
            counts[n]++;
            if(counts[n] == 2){
                duplicate++;
            }
        }
        int result = 0;
        while (duplicate>0){

            for(int i=3*result;i<Math.min(nums.length, (result+1)*3);i++ ){
                int n = nums[i];
                counts[n]--;
                if(counts[n] == 1){
                    duplicate--;
                }
                
            }

            result++;
        }

        return result;
    }
}